src/Containers/FormContainer.tsx

```diff
@@ -84,21 +85,34 @@ class LoginFormContainer extends React.PureComponent<Props, State> {
           key={fieldName}
           label={field.label}
           secureTextEntry={field.secureTextEntry}
-          validateValue={field.validate}
+          validateValue={this.validateValue(fieldName)}
           onChangeTextValidated={this.setValue(fieldName)}
           onSuccessInputFieldColor={theme.themeColor}
           textColor={theme.defaultText}
           keyboardType={field.keyboardType}
           initialValue={field.initialValue}
-          onSubmitEditing={this.onCurrentInputSubmit(nextInputName)}
+          onSubmitEditing={this.handleInputSubmit(nextInputName)}
         />
       );
     });
   };
+  private validateValue = (fieldName: string) => (value?: string) => {
+    const field = this.props.fields[fieldName];
+    const preValidateFieldsState = {
+      ...this.state.fields,
+      [fieldName]: { value, isValid: false },
```

Comment:
why do we need to make `isValid` false, before sending it to validate function? Why don't we just send `this.state.fields`?

Reply:
because `this.state.fields` contain a previous value of current input value, and we don't valid it or not